packages/context/src/watcher.ts

```diff
@@ -0,0 +1,193 @@
+import { fileURLToPath } from "url";
+import { findAppRoot, getProjectInfo, getProjectRoot } from "../src/utils";
+import { readManifestFile } from "../src/manifest";
+import { getApp } from "./loader";
+import { cache } from "./cache";
+import { getManifestDetails, getUI5Manifest } from "./manifest";
+import { FileName } from "@sap-ux/project-access";
+import { join } from "path";
+import { FileChangeType } from "vscode-languageserver";
+import { URI } from "vscode-uri";
+import { readUI5YamlFile } from "./ui5-yaml";
+
+const handleManifestFileChange = async (
+  manifestPath: string
+): Promise<void> => {
+  const projectRoot = await getProjectRoot(manifestPath);
+  if (!projectRoot) {
+    return;
+  }
+  const appRoot = await findAppRoot(manifestPath);
+  if (!appRoot) {
+    return;
+  }
+  const manifest = await getUI5Manifest(manifestPath);
+  if (!manifest) {
+    return;
+  }
+  const manifestDetails = await getManifestDetails(manifestPath);
+  const projectInfo = await getProjectInfo(projectRoot);
+  // remove cached app
+  cache.deleteApp(appRoot);
+  const app = await getApp(
+    projectRoot,
+    appRoot,
+    manifest,
+    manifestDetails,
+    projectInfo
+  );
+  if (!app) {
+    return;
+  }
+  const cachedCapProject = cache.getProject(projectRoot);
+  if (cachedCapProject) {
+    if (cachedCapProject.type === "CAP") {
+      cachedCapProject.apps.set(appRoot, app);
+    } else {
+      cachedCapProject.app = app;
+    }
+  }
+};
+/**
+ * React on manifest.json file change
+ *
+ * @param uri uri to manifest.json file
+ * @description
+ * 1. for change or create operation:
+ *
+ *    a. fresh manifest entry is added to cache
+ *
+ *    b. cached entry of app is removed and a fresh app entry is added to cache
+ *
+ *    c. fresh app is assigned to respective UI5 or CAP project app(s)
+ *
+ * 2. for delete operation:
+ *
+ *     a. cached manifest entry is removed from cache
+ */
+export const reactOnManifestChange = async (
+  manifestUri: string,
+  changeType: FileChangeType
+): Promise<void> => {
+  console.debug("`reactOnManifestChange` function called", {
+    manifestUri,
+    changeType,
+  });
+  const manifestPath = URI.parse(manifestUri).fsPath;
+  switch (changeType) {
+    case 1: //created
```

Comment:
No reaction on creation? Since we handle manifest deletion, I'd expect handler for its creation